lib/components/conversation/conversation_item.dart

```diff
@@ -0,0 +1,168 @@
+      ..innerText = _id;
+
+    var messageElement = DivElement()..className = "conversation-item-message";
+    var messageTextElement = SpanElement()
+      ..className = "conversation-item-message-text"
+      ..innerText = _message;
+    _messageStatusElement = SpanElement()..className = "conversation-item-status";
+    _updateStatus(_status);
+
+    messageElement..append(messageTextElement)..append(_messageStatusElement);
+    contentWrapper..append(idElement)..append(messageElement);
+    renderElement..append(checkboxWrapper)..append(contentWrapper);
+
+    this._onChooseController = StreamController();
+    this._onChoose = _onChooseController.stream;
+    this._onSelectController = StreamController();
+    this._onSelect = _onSelectController.stream;
+    this._onDeselectController = StreamController();
+    this._onDeselect = _onDeselectController.stream;
+  }
+
+  void _updateStatus(ConversationItemStatus status) {
+    var messageStatusClasses = _messageStatusElement.classes
+      ..removeWhere((classname) {
+        return classname.indexOf("converversation-item-status--") == 0;
+      });
+    var renderElementClasses = renderElement.classes
+      ..removeWhere((classname) {
+        return classname.indexOf("conversation-item--selected") != 0 && classname.indexOf("conversation-item--") == 0;
+      });
+    switch (status) {
+      case ConversationItemStatus.draft:
+        renderElementClasses.add("conversation-item--draft");
+        renderElement.className = renderElementClasses.join(" ");
+        messageStatusClasses.add("converversation-item-status--draft");
+        _messageStatusElement
+          ..innerText = "[draft]"
+          ..className = messageStatusClasses.join(" ");
+        break;
+      case ConversationItemStatus.failed:
+        renderElementClasses.add("conversation-item--failed");
+        renderElement.className = renderElementClasses.join(" ");
+        messageStatusClasses.add("converversation-item-status--failed");
+        _messageStatusElement
+          ..innerText = "[delivery failure]"
+          ..className = messageStatusClasses.join(" ");
+        break;
+      case ConversationItemStatus.pending:
+        renderElementClasses.add("conversation-item--pending");
+        renderElement.className = renderElementClasses.join(" ");
+        messageStatusClasses.add("converversation-item-status--pending");
+        _messageStatusElement
+          ..innerText = "[pending]"
+          ..className = messageStatusClasses.join(" ");
+        break;
+      default:
+        renderElementClasses.add("conversation-item--normal");
+        renderElement.className = renderElementClasses.join(" ");
+        messageStatusClasses.add("converversation-item-status--normal");
+        _messageStatusElement
+          ..innerText = ""
+          ..className = messageStatusClasses.join(" ");
+    }
+  }
+
+  void select() {
+    _selected = true;
+    _checkboxElement.checked = true;
+  }
+
+  void unSelect() {
```

Comment:
nit, no need for capitalisation:
```suggestion
  void unselect() {
```